lib/graphql/tracing/platform_trace.rb

```diff
@@ -5,12 +5,22 @@ module Tracing
     module PlatformTrace
       def initialize(trace_scalars: false, **_options)
         @trace_scalars = trace_scalars
-        @platform_field_key_cache = Hash.new { |h, k| h[k] = platform_field_key(k) }
-        @platform_authorized_key_cache = Hash.new { |h, k| h[k] = platform_authorized_key(k) }
-        @platform_resolve_type_key_cache = Hash.new { |h, k| h[k] = platform_resolve_type_key(k) }
+
+        @platform_key_cache = Hash.new { |h, mod| h[mod] = mod::KeyCache.new }
```

Comment:
The basic gist of the problem is here: since all the modules are mixed into a single class, they share a single set of instance variables and method definitions. I addressed this problem in two ways: 

- Platform keys are now cached per-module, so each module can reference its own cached keys. 
- Platform keys aren't fetched from the trace object itself; instead, the platform trace module is mixed into a _new_ class, used only for getting those keys. That new class is assigned to `mod::KeyCache`.